src/scicat_dataset.py

```diff
@@ -54,6 +65,45 @@ def convert_to_type(input_value: Any, dtype_desc: str) -> Any:
     return converter(input_value)
 
 
+_OPERATOR_REGISTRY = MappingProxyType(
+    {
+        "DO_NOTHING": lambda value: value,
+        "join_with_space": lambda value: ", ".join(value),
+    }
+)
+
+
+def _get_operator(operator: str | None) -> Callable:
+    return _OPERATOR_REGISTRY.get(operator or "DO_NOTHING", lambda _: _)
+
+
+def extract_variables_values(
+    variables: dict[str, dict], h5file: h5py.File, config: SciCatOptions
+) -> dict:
+    variable_map = {}
+    for variable_name, variable_recipe in variables.items():
+        if (source := variable_recipe["source"]) == "NXS":
+            value = h5file[variable_recipe["path"]][...]
+        elif source == "SC":
+            value = retrieve_value_from_scicat(
+                config=config,
+                variable_url=render_variable_value(
+                    variable_recipe["url"], variable_map
+                ),
+                field_name=variable_recipe["field"],
+            )
+        elif source == "VALUE":
+            value = _get_operator(variable_recipe.get("operator"))(
+                render_variable_value(variable_recipe["value"], variable_map)
+            )
+        else:
+            raise Exception("Invalid variable source: ", source)
+        variable_map[variable_name] = convert_to_type(
+            value, variable_recipe["value_type"]
+        )
+    return variable_map
+
+
 @dataclass(kw_only=True)
 class TechniqueDesc:
     pid: str
```

Comment:
Technique has an additional field named "name". Its type is string

Reply:
One more thing: techniques can be empty.

Reply:
> Technique has an additional field named "name". Its type is string

@nitrosx  is it `name` or `names`...? 
If you expand the changes, there is a field called `names`. I copied it from the previous implementation.

> One more thing: techniques can be empty.

Yes, it's handled here:
https://github.com/SciCatProject/scicat-filewriter-ingest/blob/3f3359e79ffeeb2e19e50415cc8f51f7a61263ab/src/scicat_dataset.py#L467-L469

Or you mean not-setting it shouldn't be the same as setting an empty list...?

Reply:
Dataset techniques fields is a list of dictionaries with the following structure:
```
{
 'pid' : str,
 'name' : str
}
```

If no technique is provided, the field should be an empty array.
Right now, I'm planning to leave the field empty. Future upgrades will include them. I'm still talking to the instrument teams to figure out which information they would like to have in here